### PR TITLE
Move mixed Java ITs to Shared

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
Prerequisite for #5436 

VB.NET ITs currently contains C# & VB.NET analyzer. We will need to remove the C# one from VB.NET ITs to prove the packaging. For this reason, we need to move all mixed ITs into `shared` project.

As a consequence, this PR cleans up `shared` project structure and aligns it with `csharp` and `vbnet` by using `Tests.java` with `SuiteClasses`.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,14 +1,14 @@
 schedules:
-# Run from Monday to Friday at 5:30 UTC (https://docs.microsoft.com/en-us/azure/devops/pipelines/process/scheduled-triggers?view=azure-devops&tabs=yaml#cron-syntax)
```
Comment (csaba-sagi-sonarsource) [inline azure-pipelines.yml:2]: Why this change? What is actually changed here?

Comment (pavel-mikula-sonarsource) [reply azure-pipelines.yml:2]: Seems like IntelliJ was not intelligent enough and messed up whitespace. I'll fix that. There's IT change that should pop up.

Comment (pavel-mikula-sonarsource) [reply azure-pipelines.yml]: Done in the original commit

File: azure-pipelines.yml [hunk 2]
```diff
@@ -467,7 +467,7 @@ stages:
           CSharp:
             testClasses: 'com.sonar.it.csharp.Tests'
           Others:
```
Comment (csaba-sagi-sonarsource) [inline azure-pipelines.yml:469]: Not really a comment for this PR. Why don't we have a matrix of CSharp, VBNet and Shared?

Comment (pavel-mikula-sonarsource) [reply azure-pipelines.yml:469]: Because `VBNET`+`Shared` are faster than `CSharp`. So it's fine to split them like this for now. This parallelization was done quite recently.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!
